dashboard/lti_new.py

```diff
@@ -64,17 +69,40 @@ def check_if_success_getting_tool_config(tool_config):
         logger.info('Success in fetching LTI tool configuration')
         return True
     else:
-        logger.error(tool_config)
+        logger.error(f'Invalid LTI configuration: "{tool_config}"')
         return False
 
 
-def get_jwks(request):
+def get_jwks(_):
     config = get_tool_conf()
     if not check_if_success_getting_tool_config(config):
         return LTIError(config).response_json()
     return JsonResponse(config.get_jwks())
 
 
+def generate_config_json(request: HttpRequest) -> \
+        Union[HttpResponse, LTIError]:
+    config = get_tool_conf()
+    if not check_if_success_getting_tool_config(config):
+        return LTIError(f'Invalid LTI configuration: "{config}"')\
+            .response_json()
+
+    parameters = {
+        'timestamp': datetime.now().isoformat(),
+        'host': request.get_host(),
+        'base_url': urllib.parse.urlunsplit(
+            [request.scheme, request.get_host()] + 3 * ['']),
+        'login_url_suffix': reverse('login'),
+        'launch_url_suffix': reverse('launch'),
+        'jwks_url_suffix': reverse('get_jwks'),
+    }
+
+    return HttpResponse(
+        open(finders.find(
+            'config/lti_config_template.json')).read() % parameters,
+        content_type='application/json')
```

Comment:
`open(finders.find(` this statement seems to directly opening the file and making an assumption the we will find that file there and exception handling/logging are not in place? 
Just before rendering as HTTPResponse we need to check if file exists and params are rightly applied and log accordingly stating the issue

Reply:
Yes, it does use the file without checking for errors.

If you think it's necessary to check for errors reading the file, I can add that.  I didn't think it was absolutely necessary because the file is being read from the project folders, as suggested by @jonespm.  That is, we have more control over the file than if it were in `/secrets`.